validator/client/validation_client.go

```diff
@@ -153,10 +153,14 @@ func NewExecutionClient(config rpcclient.ClientConfigFetcher, stack *node.Node)
 	}
 }
 
-func (c *ExecutionClient) CreateExecutionRun(wasmModuleRoot common.Hash, input *validator.ValidationInput) containers.PromiseInterface[validator.ExecutionRun] {
-	return stopwaiter.LaunchPromiseThread[validator.ExecutionRun](c, func(ctx context.Context) (validator.ExecutionRun, error) {
+func (c *ExecutionClient) CreateExecutionRun(
+	wasmModuleRoot common.Hash,
+	input *validator.ValidationInput,
+	useBoldMachine *bool,
```

Comment:
I think we can just use a bool without the pointer here. The reason for the pointer is so that the JSON-RPC interface is backwards compatible, but that only matters on the server side not the client side. Then again, if the pointer is necessary here to implement an interface, that's fine.